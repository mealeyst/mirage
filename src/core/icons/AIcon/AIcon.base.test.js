import React from 'react'
import 'jest-styled-components'

import BaseAIcon, { AIcon } from './AIcon.base'

const { shallowWithTheme } = global

describe('(Styled component) BaseAIcon', () => {
  const createBaseAIcon = (props) => {
    return shallowWithTheme(<BaseAIcon {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createBaseAIcon())
    .toMatchSnapshot()
  })

  test('setting the width', () => {
    const width = '4rem'
    expect(createBaseAIcon({width: width}))
    .toHaveStyleRule({
      width: width
    })
  })
})

describe('(Component) AIcon', () => {
  const createAIcon = (props) => {
    return shallowWithTheme(<AIcon {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createAIcon())
    .toMatchSnapshot()
  })

  test('setting the className', () => {
    const className = 'example-class'
    expect(
      createAIcon({className: className}).prop('className')
    ).toEqual(className)
  })
})
