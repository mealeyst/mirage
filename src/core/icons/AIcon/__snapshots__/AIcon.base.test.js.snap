// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) AIcon matching the snapshot 1`] = `
<svg
  viewBox="0 0 80 80"
>
  <path
    d="M57.5,64.6V40.1c0-9.4,0.1-14.8-4.7-19.5c-7.2-7-18.5-7-25.6,0c-4.8,4.8-4.8,10.2-4.8,19.5v24.5h7.4V50.3h20.2 v14.4H57.5z M29.9,43.6V40c0-9.8,0.5-12.5,2.8-15.1c3.9-4,10.5-4,14.5,0l0,0C49.6,27.6,50,30.3,50,40v3.5H29.9z"
  />
</svg>
`;

exports[`(Styled component) BaseAIcon matching the snapshot 1`] = `
.c0 {
  width: 9.2rem;
}

<AIcon
  className="c0"
  width="9.2rem"
/>
`;
